.github/workflows/runtime-image-update.yaml

```diff
@@ -0,0 +1,105 @@
+        id: get_latest_tag
+        # yamllint disable rule:line-length
+        run: |
+          LATEST_TAG=$(python ./.github/scripts/get_latest_runtime_image_tag.py ${{ github.event.inputs.quay_repo }})
+          echo "LATEST_TAG=$LATEST_TAG"
+          echo "::set-output name=tag::$LATEST_TAG"
+        # yamllint enable rule:line-length
+
+      - name: Check if tag was updated
+        id: check_tag_updated
+        run: |
+          if [ "${{ steps.get_current_tag.outputs.tag }}" != "${{ steps.get_latest_tag.outputs.tag }}" ]; then
+            echo "Tag was updated"
+            echo "::set-output name=tag_updated::true"
+          else
+            echo "Tag was not updated"
+            echo "::set-output name=tag_updated::false"
+          fi
+
+      - name: Create runtime image tag update branch
+        id: create_runtime_image_update_branch
+        run: |
+          BRANCH_NAME="update-runtime-image-tag-${{ steps.get_latest_tag.outputs.tag }}"
+          git fetch origin main
+          git checkout main
+          git checkout -b $BRANCH_NAME
+          echo "BRANCH_NAME=$BRANCH_NAME"
+          echo "runtime_image_update_branch=$BRANCH_NAME" >> $GITHUB_OUTPUT
+
+      - name: Replace tag in Dockerfile.astro_cloud
+        if: steps.check_tag_updated.outputs.tag_updated == 'true'
+        # yamllint disable rule:line-length
+        run: |
+          sed -i "s@quay.io/astronomer/astro-runtime:${{ steps.get_current_tag.outputs.tag }}@quay.io/astronomer/astro-runtime:${{ steps.get_latest_tag.outputs.tag }}@g" ./.circleci/integration-tests/Dockerfile.astro_cloud
+          echo "::group::cat Dockerfile.astro_cloud"
+          cat  ./.circleci/integration-tests/Dockerfile.astro_cloud
+          echo "::endgroup::"
+        # yamllint enable rule:line-length
+
+      - name: Replace tag in Makefile
+        if: steps.check_tag_updated.outputs.tag_updated == 'true'
+        # yamllint disable rule:line-length
+        run: |
+          sed -i "s@quay.io/astronomer/astro-runtime:${{ steps.get_current_tag.outputs.tag }}@quay.io/astronomer/astro-runtime:${{ steps.get_latest_tag.outputs.tag }}@g" ./Makefile
+          echo "::group::cat Makefile"
+          cat ./Makefile
+          echo "::endgroup::"
+        # yamllint enable rule:line-length
+
+      - name: Setup Github Actions git user
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Actions"
+
+      - name: Commit changes and create a pull request
+        env:
+          GH_TOKEN: ${{ github.token }}
+        # yamllint disable rule:line-length
+        run: |
+          set -e
+          git add Makefile .circleci/integration-tests/Dockerfile.astro_cloud
+          git commit -m "Update runtime image tag to ${{ steps.get_latest_tag.outputs.tag }}"
+          git push origin ${{ steps.create_runtime_image_update_branch.outputs.runtime_image_update_branch }}
+          gh pr create --base main \
```

Comment:
Should we check whether the pr has been created? Otherwise, wouldn't it generate a PR per day if we do not update the image in time? If this is needed in this automation as well, I have implemented similar behavior [here](https://github.com/astronomer/astronomer-providers/pull/1146/files#diff-7cd74bfb9babddfbfa7911a72e5d0784f450360d0a81e9f8319ecba30711b059R64).

Reply:
Thanks for this. I was missing another condition for these steps to check whether the tag has been updated, added that now so that if at least the main contains the latest tag, we do not proceed.

Coming back to your question, if we forget to update the image in time (which should not happen ideally) since the branch name would be the same, this step would fail when it tries to push the changes saying that upstream has changes that need to be pulled in first from the same branch. So we're safe that way. I agree, yours is the cleaner way but let's see if I find time later, will make that change too.
e.g. an action run when it failed for the scenario we're discussing https://github.com/astronomer/astronomer-providers/actions/runs/5155815512/jobs/9285972413